client/jetpack-connect/controller.js

```diff
@@ -106,6 +107,11 @@ export function redirectWithoutLocaleIfLoggedIn( context, next ) {
 
 export function maybeOnboard( { query, store }, next ) {
 	if ( ! isEmpty( query ) && query.onboarding ) {
+		if ( versionCompare( query.jp_version, '5.8', '<' ) ) {
```

Comment:
We should be careful about using query args without checking if they exist, because they're coming from a query string after all... Should we protect the `query. jp_version ` usage with a `get()`?

Reply:
A `get()` won't suffice tho, right? If we can't find out the version number (and `site_url`, below), we can't redirect (or do a lot at all, really). But yeah, I can add some guards.

Reply:
Yeah, you're right. But still, some guards will prevent Calypso from failing fatally, which is what we want to avoid - it's more than clear that something is very wrong if the version/url are missing from the auth query string.